dftimewolf/lib/state.py

```diff
@@ -260,24 +260,47 @@ def GetStats(self) -> List[StatsEntry]:
 
   def GetContainers(self,
                     container_class: Type[T],
-                    pop: bool=False) -> Sequence[T]:
+                    pop: bool=False,
+                    metadata_filter_key: Optional[str]=None,
+                    metadata_filter_value: Optional[Any]=None) -> Sequence[T]:
     """Thread-safe method to retrieve data from the state's store.
 
     Args:
       container_class (type): AttributeContainer class used to filter data.
       pop (Optional[bool]): Whether to remove the containers from the state when
           they are retrieved.
+      metadata_filter_key (Optional[str]): Metadata key to filter on.
+      metadata_filter_value (Optional[Any]): Metadata value to filter on.
 
     Returns:
       Collection[AttributeContainer]: attribute container objects provided in
           the store that correspond to the container type.
+
+    Raises:
+      RuntimeError: If only one metadata filter parameter is specified.
     """
+    if bool(metadata_filter_key) ^ bool(metadata_filter_value):  # logical XOR
```

Comment:
Maybe find a more Pythonic way of checking this, something like `if None in (metadata_filter_key, metadata_filter_value).

Reply:
Disregard this suggestion, I misunderstood what you're trying to do with this line.